src/system/SystemLayerImplSelect.cpp

```diff
@@ -184,7 +192,12 @@ CHIP_ERROR LayerImplSelect::StartTimer(Clock::Timeout delay, TimerCompleteCallba
     ev_timer_init(&timer->mLibEvTimer, &LayerImplSelect::HandleLibEvTimer, 1, 0);
     timer->mLibEvTimer.data = timer;
     auto t                  = Clock::Milliseconds64(delay).count();
-    ev_timer_set(&timer->mLibEvTimer, static_cast<double>(t) / 1E3, 0.);
+    // Note: libev uses the time when events started processing as the "now" reference for relative timers,
+    //   for efficiency reasons. This point in time is represented by ev_now().
+    //   The real time is represented by ev_time().
+    //   As some chip code relies on timers to fire NEVER even only a bit early, we must increase the
```

Comment:
That code is buggy and needs to be fixed.  See https://github.com/project-chip/connectedhomeip/issues/28929

Reply:
I changed the comment, but kept the adjustment?

While the fix should not be necessary as per #28929, systematically firing all timers a bit too early is not nice.

Reply:
@plan44 Sure, that's why I approved.